lib/models/addon-discovery.js

```diff
@@ -97,6 +97,15 @@ AddonDiscovery.prototype.discoverFromDependencies = function(root, pkg, excludeD
       if (addonPath) {
         return discovery.discoverAtPath(addonPath);
       }
+
+      // this supports packages that do not have a valid entry point
+      // script (aka `main` entry in `package.json` or `index.js`)
+      addonPath = path.join(root, 'node_modules', name);
+      var addon = discovery.discoverAtPath(addonPath);
+      if (addon) {
+        console.warn('The package `' + name + '` is not a properly formatted package, we have used a fallback lookup to resolve it at `' + addonPath + '`. This is generally caused by an addon not having a `main` entry point (or `index.js`).');
```

Comment:
Merging this for now, to get master properly working again with ember-data, and will PR a refactor adding `ui` to `Project` and `Addon` separately (it may come in handy in other places also).